src/components/clusterDetail/ClusterProgress.tsx

```diff
@@ -220,36 +254,39 @@ const ClusterProgress: React.FC<ClusterProgressProps> = ({ cluster, minimizedVie
           </FlexItem>
         </Flex>
       </DetailList>
-      {!minimizedView && (
-        <>
-          <Progress
-            value={progressPercent}
-            label={label}
-            title=" "
-            measureLocation={getMeasureLocation(status)}
-            variant={getProgressVariant(status)}
-            className="cluster-progress-bar"
-          />
-          <Flex className="pf-u-mt-md" display={{ default: 'inlineFlex' }}>
+      <RenderIf condition={!minimizedView}>
+        <Progress
+          value={hostsProgressPercent}
+          label={label}
+          title=" "
+          measureLocation={getMeasureLocation(status)}
+          variant={getProgressVariant(status)}
+          className="cluster-progress-bar"
+        />
+        <Flex className="pf-u-mt-md" display={{ default: 'inlineFlex' }}>
+          <FlexItem>
+            <HostProgress hosts={enabledHosts} hostRole="master" />
+          </FlexItem>
+
+          <RenderIf condition={isWorkersPresent}>
             <FlexItem>
-              <HostProgress hosts={hosts} hostRole="master" />
+              <HostProgress hosts={enabledHosts} hostRole="worker" />
             </FlexItem>
-            {isWorkersPresent && (
-              <FlexItem>
-                <HostProgress hosts={hosts} hostRole="worker" />
-              </FlexItem>
-            )}
+          </RenderIf>
+
+          <FlexItem>
+            <FinalizingProgress cluster={cluster} />
+          </FlexItem>
+
+          <RenderIf condition={monitoredOperators.length > 0}>
```

Comment:
This progress item should render only if there are any olm operators being installed.
```suggestion
          <RenderIf condition={getOlmOperators(monitoredOperators).length > 0}>
```